enterprise/frontend/src/embedding/data-picker/DataSelector/DataSelector.jsx

```diff
@@ -366,16 +369,29 @@ export class UnconnectedDataSelector extends Component {
     } else if (
       // Schema id is explicitly set when going through the New > Question/Model flow,
       // whereas we have to obtain it from the state when opening a saved question.
-      (this.state.selectedSchemaId || this.state.selectedSchema?.id) &&
-      steps.includes(TABLE_STEP)
+      this.state.selectedSchemaId ||
+      this.state.selectedSchema?.id
     ) {
       await this.switchToStep(TABLE_STEP);
-    } else if (this.state.selectedDatabaseId && steps.includes(SCHEMA_STEP)) {
-      await this.switchToStep(SCHEMA_STEP);
-    } else if (steps[0] === DATA_BUCKET_STEP && !this.hasUsableModels()) {
-      await this.switchToStep(steps[1]);
+    } else if (this.isJoinStep()) {
+      const isQuerySourceModel = this.props.isQuerySourceModel;
+
+      if (isQuerySourceModel) {
+        await this.switchToStep(
```

Comment:
perform the same step when selecting the `Model` bucket.
https://github.com/metabase/metabase/blob/f7a24711c48f957636ec6e5af0e37a0a9df20cc3/enterprise/frontend/src/embedding/data-picker/DataSelector/DataSelector.jsx#L608-L614
![image](https://github.com/user-attachments/assets/89bbd1a9-6f06-4240-b251-a9ede7e220a7)